src/site/markdown/index.md

```diff
@@ -19,102 +19,110 @@
-should give noticeable benefits in medium and small sized projects as well.
+should also give noticeable benefits in medium and small-sized projects.
 
 ### Cache concepts
 
-The idea of the build cache is to calculate key from module inputs, store outputs in cache and restore them later
-transparently to the standard Maven core. In order to calculate the key cache engine analyzes source code, build flow,
-plugins and their parameters. This allows to deterministically associate each project state with unique key and restore
-up-to-date (not changed) projects from cache and rebuild out-of-date(changed) ones. Restoring artifacts associated with
-a particular project state improves build times by avoiding re-building unnecessary modules. Cache does not make any
-interventions to actual build execution process and fully delegates build work to Maven core. This ensures that
-artifacts produced in presence of cache are equivalent to result produced by a standard Maven build.   
-To achieve accurate key calculation incremental Maven combines automatic introspection
-of [project object model](https://maven.apache.org/pom.html#What_is_the_POM) and allows fine-grained tuning by means of
-configuration file and xml attributes. Source code content fingerprinting is digests based which is more reliable over
-widely used file timestamps in tools like Make or Apache Ant. Deterministic build state allows reliably cache outputs
-even of the build in progress and share them between teams using remote cache. Deterministic inputs calculation allows
-distributed and parallel builds running in heterogeneous environments (like cloud of build agents)
-could efficiently reuse cached build artifacts. Therefore, incremental Maven is particularly well-suited for large Maven
-projects that have significant number of small modules. Remote cache in conjunction with relocatable inputs
-identification effectively enables "change once - build once" approach across all environments.
+The idea of the build cache is to calculate a key from module inputs, store outputs in the cache, and restore them later
+transparently to the standard Maven core. The cache deterministically associates each project state with a unique key
+and restores it in subsequent builds. It analyzes source code, project model,
+plugins, and their parameters. Projects with the same key are up-to-date (not changed) and could be safely restored from
+the cache. Projects producing different keys are out-of-date (changed), and the cache fully rebuilds them. In the latter
+case, the cache does not make any
+interventions to the build execution logic and delegates build work to the standard maven Maven core. This approach
+ensures that
+artifacts produced in the presence of a cache are equivalent to the result produced by a standard Maven build.   
+To achieve an accurate key calculation, incremental Maven combines automatic introspection
```

Comment:
`incremental Maven` -> `the build-cache extension`

Reply:
fixed